### PR TITLE
Lets make sure we are waiting for search results

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

Switch search to use a for loop instead of a while.

## How to test

All tests should pass.
Make sure we get back the correct number of results from 'Report text' search.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-0

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

**Comment by GarrettEHill** [comment]
I don't see how this is effectively different other then changing the max retry from 9 to 5

**Comment by AdamAdHocTeam** [comment]
> I don't see how this is effectively different other then changing the max retry from 9 to 5

In JavaScript a WHILE loop will not wait for the async call to come back. This might lead to some strange overlapping requests. The FOR loop ensures the response comes back before it requests the next batch or results (if there are any).

File: src/lib/awsElasticSearch/index.js [hunk 1]
```diff
@@ -180,8 +177,9 @@ const search = async (indexName, fields, query, passedClient, overrideBatchSize)
         },
       ],
     };
-
-    while (retrieveAgain && loopIterations <= maxLoopIterations) {
+    // Use a FOR loop to force synchronous requests.
```
Comment (kryswisnaskas) [inline src/lib/awsElasticSearch/index.js:180]:
```suggestion
```

File: src/lib/awsElasticSearch/index.js [hunk 2]
```diff
@@ -180,8 +177,9 @@ const search = async (indexName, fields, query, passedClient, overrideBatchSize)
...
+    // This will help us keep the ES queue down.
```
Comment (kryswisnaskas) [inline src/lib/awsElasticSearch/index.js:181]:
```suggestion
```

**Review by kryswisnaskas — APPROVED** [review]
Works locally!

**Comment by AdamAdHocTeam** [comment]
This is not needed, as long as a the while statement is in a async function it will work as expected.

**Pull request closed without merging**